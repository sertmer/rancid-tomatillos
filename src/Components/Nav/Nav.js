import React from 'react';
import './Nav.scss';
import Login from '../Login/Login';

const Nav = () => {
  return(
    <header>
      <img  />
      <h1>Rancid Tomatillos</h1>
      <Login />
    </header>
  )
}

export default Nav;
