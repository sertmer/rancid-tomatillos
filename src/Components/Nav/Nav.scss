@import '../../scss/mixins';
@import '../../scss/variables';

header {
  @include rounded-corners(50px);
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.rancid-title {
  @include rounded-corners(50px);
  margin: 1.5% 1.5% 1.5% 31%;
  background: rgb(224,224,224);
  background: linear-gradient(
    0deg, 
    rgba(224,224,224,1) 0%, 
    rgba(245,245,245,1) 100%
  );
  display: flex;
  justify-content: center;
  width: 15vw;
  height: 7vh;
  padding-top: 15px;
}

.login-button {
  width: 10vw;
  height: 5vh;
  @include rounded-corners(5px);
  background-color: $green;
  margin-top: 27%;
  margin-left: 76%;
  font-family: $accent-font;
  font-size: larger;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.login-button:hover {
  transform: scale(1.1);
}