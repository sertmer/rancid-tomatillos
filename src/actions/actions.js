export const getMovies = movies => ({
  type: 'GET_MOVIES',
  movies
});

export const currentUser = user => ({
  type: 'SET_CURRENT_USER',
  name: user.user.name,
  id: user.user.id,
  ratings: []
});

export const loggedIn = loggedIn => ({
  type: 'LOG_IN',
  loggedIn
});

export const getCurrentMovie = movie => ({
  type: 'GET_MOVIE',
  movie
});

export const getRatings = ratings => ({
  type: 'GET_RATINGS',
  ratings
});

