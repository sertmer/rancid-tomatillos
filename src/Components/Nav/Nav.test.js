import React from 'react';
import { Nav } from './Nav';
import { shallow } from 'enzyme';
import { logOut, mapState, mapDispatch } from '../Nav/Nav';
import { getRatings, loggedIn } from '../../actions/actions';

describe('Nav', () => {

  let wrapper

  describe('Nav Component', () => {

    beforeEach(() => {
      wrapper = shallow(<Nav
        logOut={logOut}
        isLoggedIn={true}
        login={true}
        clearRatings={jest.fn()}
      />)
    });

    it('should match the snapshot', () => {
      expect(wrapper.debug()).toMatchSnapshot()
    })
  });

  describe('mapState', () => {

    it('should return an object with isLoggedIn', () => {
      const mockState = {
        loggedIn: true,
      };
      const expected = { isLoggedIn: true };

      const mappedProps = mapState(mockState);

      expect(mappedProps).toEqual(expected);
    });

  });

  describe('mapDispatch', () => {

    it('should call mapDispatch with a login', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = loggedIn(true);

      const mappedProps = mapDispatch(mockDispatch);
      mappedProps.login(true);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call mapDispatch with no ratings', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getRatings([]);

      const mappedProps = mapDispatch(mockDispatch);
      mappedProps.clearRatings([]);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

  });



})
