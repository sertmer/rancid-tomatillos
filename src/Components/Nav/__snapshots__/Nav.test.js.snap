// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
"<header>
  <h1 className=\\"rancid-title\\">
    Rancid
  </h1>
  <Link to=\\"/login\\">
    <button className=\\"login-button\\" type=\\"button\\">
      Login
    </button>
  </Link>
</header>"
`;
